examples/example_container_vulnerabilities_csv.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+from laceworksdk import LaceworkClient
+from datetime import datetime, timedelta
+import csv
+import json
+
+# setup lacework cli (~/.lacework.toml)
+lw = LaceworkClient()
+
+start_time_dt = datetime.now() + timedelta(days=-1)
+end_time_dt = datetime.now()
+start_time = start_time_dt.strftime("%Y-%m-%dT%H:%M:%SZ")
+end_time = end_time_dt.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+query = {
+    "timeFilters": {
+        "startTime": start_time,
+        "endTime": end_time
+    },
+    "filters": [
+        {"field": "severity", "expression": "in", "values": ["Critical", "High", "Medium"]},
+        {"field": "status", "expression": "in", "values": ["VULNERABLE"]},
+        {"field": "fixInfo.fix_available", "expression": "eq", "value": 1}
+    ],
+    "returns": [
+        "startTime",
+        "imageId",
+        "severity",
+        "status",
+        "vulnId",
+        "evalCtx",
+        "fixInfo",
+        "featureKey"
+    ]
+}
+
+print("Query: {0}".format(json.dumps(query, indent=4)))
+
+with open('container_vulnerability_summary.csv', 'w') as f:
+    header = False
+    write = csv.writer(f, quoting=csv.QUOTE_ALL)
+
+    container_vulns = lw.vulnerabilities.containers.search(json=query)
+
+    for h in container_vulns:
+        for d in h['data']:
+            # create the data row
+            try:
+                row = {
+                    "startTime": d['startTime'],
+                    "imageId": d['imageId'],
+                    "cve_id": d.get('vulnId'),
+                    "package_name": d['featureKey']['name'],
+                    "package_namespace": d['featureKey']['namespace'],
+                    "version": d['featureKey']['version'],
+                    "fix_available": d['fixInfo']['fix_available'],
+                    "fixed_version": d['fixInfo']['fixed_version'],
+                    "severity": d['severity'],
+                    "status": d['status'],
+                    "image_created_time": d['evalCtx']['image_info'].get('created_time'),
+                    "image_digest": d['evalCtx']['image_info']['digest'],
+                    "image_error_msg": d['evalCtx']['image_info']['error_msg'],
+                    "image_registry": d['evalCtx']['image_info']['registry'],
+                    "image_repo": d['evalCtx']['image_info']['repo'],
+                    "image_size": d['evalCtx']['image_info']['created_time'],
+                    "image_type": d['evalCtx']['image_info']['type'],
+                    "scan_status": d['evalCtx']['image_info']['status'],
+                    "scan_start_time": d['evalCtx']['scan_request_props']['scan_start_time'],
+                    "scan_complete_time": d['evalCtx']['scan_request_props']['scanCompletionUtcTime'],
+                    "vuln_created_time": d['evalCtx']['vuln_created_time'],
+                    "tags": d['evalCtx']['image_info']['tags']
```

Comment:
Rather than cherry-picking values, would we want to just extract everything that's returned?  If we extract the values dynamically, it'll also inherit values if the returned dataset changes - or we could even filter for specific keys if you want.   We could have a function like this:
```
def value_extraction(result, parent_object, parent_key=""):

    if isinstance(parent_object, list):
        for item in parent_object:
            value_extraction(result, item, parent_key)

    if isinstance(parent_object, dict):
        for key, value in parent_object.items():
            temp_key = ".".join([parent_key, key]).strip('.')
            if isinstance(value, (dict, list)):
                value_extraction(result, value, temp_key)
            else:
                result[temp_key] = value
```

Reply:
Yeah agreed it's a pretty rigid. My original thought was to flatten everything and leave it there but nested array of tags made me think otherwise. It would be nice to support the concept of defining the field mapping, that might help with normalizing.

something like:
`fields = {
         "my_key_name": "lacework_data_path",
         etc
}` 

then as the results are returned iterate through results and output mapped row. 

one other tangentially related issue is right now there's a bit of a pain with the machine vuln data in that the hostname in the `machineTags` returned by the vuln report is different than the hostname value in machine entities result. there's some ugly dictionary building and mapping to _solve_ that issue in here as well. similar requirement comes up for finding **active** containers in the result set, need to pull container list to determine what's been _online_. 

my goal was to provide some working examples that solved the problems I know about but maybe these examples need to be simplified to handle more generic needs or maybe break out simple examples for other use cases?